renku/core/commands/providers/renku.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku dataset provider."""
+import attr
+
+from renku.core import errors
+from renku.core.commands.providers.api import ExporterApi, ProviderApi
+from renku.core.models.git import GitURL
+
+
+@attr.s
+class RenkuProvider(ProviderApi):
+    """Renku API provider."""
+
+    is_doi = attr.ib(default=False)
+    _accept = attr.ib(default='application/json')
+
+    @staticmethod
+    def supports(uri):
+        """Whether or not this provider supports a given uri."""
+        return 'renku' in uri.lower()
+
+    def find_record(self, uri, client):
+        """Retrieves a dataset from Renku.
+
+        :raises: ``LookupError``
+        :param uri: URL
+        :return: ``DataverseRecord``
+        """
+        from renku.core.management import LocalClient
+
+        remote_url = self.get_project_url(uri)
+        try:
+            GitURL.parse(remote_url)
+        except errors.ConfigurationError:
+            raise errors.ParameterError(
+                'Invalid Renku dataset URI: {}'.format(uri)
+            )
+        repo, repo_path = client.prepare_git_repo(remote_url)
+        dataset_name = self.record_id(uri)
+
+        remote_client = LocalClient(repo_path)
+        dataset = remote_client.load_dataset(dataset_name)
+
+        if dataset is None:
+            raise errors.ParameterError(
+                'Cannot find dataset with name "{}"'.format(dataset_name)
+            )
+
+        return RenkuRecordSerializer(dataset)
+
+    def get_exporter(self, dataset, access_token):
+        """Create export manager for given dataset."""
+        raise NotImplementedError()
+
+    def is_git_based(self):
+        """True if provider is git-based."""
+        return True
+
+    @staticmethod
+    def record_id(uri):
+        """Extract dataset id from uri."""
+        return uri.split(':')[-1]
```

Comment:
we need something more elaborate here to support git URLs like `git@<renku>:<user>/<project>`. Maybe we need to specify the dataset name with a flag? E.g. 

`renku dataset import --name mydata git@renkulab.io/<name>/<project>`

Reply:
The problem with git@ is not because of this. The template method where it checks if the URL is valid before handing it over to the providers and since it is not a valid URL that check fails. I'll fix it in the template method and keep the syntax as it is.
As we discussed last week in the design call, we are going to change this to be a valid URL (when UI can return JSON metadata that points to the location of the gitlab project).

Reply:
right - I just think it's a bit dangerous to add a regex that is not actually a valid git URL to the function that checks if git URLs are valid... so I would prefer a different solution even if it is "ugly" for the time being.